dds/DCPS/XTypes/DynamicDataFactory.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include <DCPS/DdsDcps_pch.h>
+
+#ifndef OPENDDS_SAFETY_PROFILE
+#  include "DynamicDataFactory.h"
+
+#  include "DynamicDataImpl.h"
+
+#include <ace/Singleton.h>
+
+using namespace OpenDDS::DCPS;
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+namespace DDS {
+
+DynamicDataFactory_ptr DynamicDataFactory::get_instance()
+{
+  return ACE_Singleton<DynamicDataFactory, ACE_SYNCH_MUTEX>::instance();
+}
+
+DynamicData_ptr DynamicDataFactory::create_data(DynamicType_ptr type)
+{
+  return new OpenDDS::XTypes::DynamicDataImpl(type);
+}
+
+DynamicDataFactory_ptr DynamicDataFactory::_duplicate(DynamicDataFactory_ptr obj)
+{
+  if (obj) {
+    obj->_add_ref();
+  }
+  return obj;
+}
+
+} // namespace DDS
+OPENDDS_END_VERSIONED_NAMESPACE_DECL
+
+TAO_BEGIN_VERSIONED_NAMESPACE_DECL
+namespace TAO {
+
+DDS::DynamicDataFactory_ptr Objref_Traits<DDS::DynamicDataFactory>::duplicate(DDS::DynamicDataFactory_ptr p)
+{
+  return DDS::DynamicDataFactory::_duplicate(p);
+}
+
+void Objref_Traits<DDS::DynamicDataFactory>::release(DDS::DynamicDataFactory_ptr)
```

Comment:
Since duplicate (just above this) seems to increment the refcount, shouldn't this decrement it?